numba/cuda/api.py

```diff
@@ -85,6 +85,13 @@ def is_cuda_array(obj):
     return hasattr(obj, '__cuda_array_interface__')
 
 
+def is_fp16_supported(device_id=None):
```

Comment:
> Thanks for the PR. There are some comments on the diff.
> 
> In addition to the comments on the diff, I think the documentation needs to be clear that this represents whether Numba supports float16 in the current configuration. If the user needs to know about the capabilities of the device they're using, they themselves should check whether the compute capability is at least 5.3 (e.g. with `cuda.get_current_device().compute_capability >= (5, 3)`). The rationale for this (as discussed OOB previously) is to avoid duplicating the purpose of the compute capability in Numba by carrying around a set of checks for particular features of the hardware.

@gmarkall the more I think about this I disagree. The attribute `cuda.get_current_device().compute_capability` is implemented in Numba, why not just add it to the `is_float16_supported method`? We are already "carrying around" this code and I am concerned developers may forget to add the additional check for the compute capability/

Reply:
I could live with an additional property of a device that indicates its support for `float16`, e.g. so you could do:

```python
# Check whether Numba supports float16 in the current configuration
cuda.is_float16_supported()
# Check whether the current device supports float16
cuda.get_current_device().supports_float16
```

where `supports_float16` is a property that returns `self.compute_capability >= (5, 3)`. I'm not keen on trying to shoehorn the check of both the bindings and the device capability into one method, because if it returns `False` there are two possible reasons why it's not supported, and it won't be clear which of them are the case.

A developer would have to check both things. I think trying to give them a single function to call in an attempt to provide a simple interface actually creates a more complicated one because it promotes ambiguity.